File: src/test_posearray.py
#!/usr/bin/env python
import rospy
import tf
import numpy as np
from geometry_msgs.msg import Pose
from geometry_msgs.msg import PoseArray

if __name__ == "__main__":

  rospy.init_node('test_posearray')
  pub = rospy.Publisher('poses_arrays', PoseArray, queue_size=1)
  rate = rospy.Rate(10)

  while not rospy.is_shutdown():

    pose_array = PoseArray()
    pose_array.header.frame_id = "/base_link"
    pose_array.header.stamp = rospy.Time.now()

    pose = Pose()
    pose.position.x = 2
    pose.position.y = 2
    pose.position.z = 0
    pose.orientation.x = 0
    pose.orientation.y = 0
    pose.orientation.z = .7071
    pose.orientation.w = .7071

    pose_array.poses.append(pose)

    pose = Pose()
    pose.position.x = 1
    pose.position.y = 1
    pose.position.z = 0
    pose.orientation.x = 0
    pose.orientation.y = 0
    pose.orientation.z = 0
    pose.orientation.w = 1

    pose_array.poses.append(pose)

    pub.publish(pose_array)

    rate.sleep()

File: src/test_posearray.cpp
#include "ros/ros.h"
#include "geometry_msgs/Pose.h"
#include "geometry_msgs/PoseArray.h"

#include <sstream>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "test_posearray");
  ros::NodeHandle n;
  ros::Publisher pub = n.advertise<geometry_msgs::PoseArray>("poses_arrays", 1);
  ros::Rate loop_rate(10);

  while (ros::ok())
  {
    geometry_msgs::PoseArray parray;

    geometry_msgs::Pose pose0;
    pose0.position.x = 2.0;
    pose0.position.y = 2.0;
    pose0.orientation.z = 0.7071;
    pose0.orientation.w = 0.7071;
    parray.poses.push_back(pose0);

    geometry_msgs::Pose pose1;
    pose1.position.x = 1.0;
    pose1.position.y = 1.0;
    pose1.orientation.z = 0;
    pose1.orientation.w = 1;
    parray.poses.push_back(pose1);

    pub.publish(parray);
    ros::spinOnce();
    loop_rate.sleep();
  }

  return 0;
}